raster/r.texture/testsuite/test_texture.py

```diff
@@ -23,60 +28,192 @@ def setUpClass(cls):
-        self.assertRasterFitsUnivar(corr_output, reference=string, precision=2)
-
-    def test_sa(self):
-        """Testing method sa"""
-        sa_SA = "sa_SA"
-        string = """min=95.95
-        max=2942.79
-        cells=1000246"""
-        self.assertModule("r.texture", input=self.input, output="sa", method="sa")
-        self.assertRasterFitsUnivar(sa_SA, reference=string, precision=2)
+        basename = "Corr"
+        method = "corr"
+        output = f"{method}_{basename}"
+        values = """min=-0.533553719520569
+        max=0.443134188652039
+        mean=-0.0511981885382368
+        variance=0.0306650179549719
+        n=996244"""
+        self.assertModule("r.texture", input=self.input, output=method, method=method)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
 
     def test_var(self):
         """Testing method var"""
-        var_output = "var_Var"
-        self.assertModule("r.texture", input=self.input, output="var", method="var")
-        self.assertRasterMinMax(
-            map=var_output,
-            refmin=0,
-            refmax=9843.07,
-            msg="var_Var in degrees must be between 0.74 and 9843.07",
-        )
+        basename = "Var"
+        method = "var"
+        output = f"{method}_{basename}"
+        values = """min=0
+        max=8776.787109375
+        mean=90.3218668390769
+        variance=58439.1071892438
+        n=996244"""
+        self.assertModule("r.texture", input=self.input, output=method, method=method)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
 
     def test_idm(self):
         """Testing method idm"""
-        idm_IDM = "idm_IDM"
-        self.assertModule("r.texture", input=self.input, output="idm", method="idm")
-        self.assertRasterMinMax(
-            map=idm_IDM,
-            refmin=0,
-            refmax=9843.07,
-            msg="idm_IDM_IDM in degrees must be between 0.74 and 9843.07",
-        )
+        basename = "IDM"
+        method = "idm"
+        output = f"{method}_{basename}"
+        values = """min=0.000490661768708378
+        max=1
+        mean=0.12308521457551
+        variance=0.00497110161251246
+        n=996244"""
+        self.assertModule("r.texture", input=self.input, output=method, method=method)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
+
+    def test_sa(self):
+        """Testing method sa"""
+        basename = "SA"
+        method = "sa"
+        output = f"{method}_{basename}"
+        values = """min=95.9583358764648
+        max=2942.79150390625
+        mean=742.367934271396
+        variance=31616.3527726298
+        n=996244"""
+        self.assertModule("r.texture", input=self.input, output=method, method=method)
+        self.assertRasterFitsUnivar(output, reference=values, precision=1e-2)
+
+    @unittest.skipIf(IS_MAC, reason="Fails on macOS")
```

Comment:
One more thing, it will be better if this says "Produces a slightly different result on macOS"

Reply:
Could we have a second test only for macOS+arm with a different precision until we find the issue? (Testing the same thing)

Reply:
I think using a very large precision would be confusing. Maybe we could use the other set of values for MacOS. @wenzeslaus Do you think we should have another test for macOS or just skip it now?

Reply:
No strong opinion, either one is good with me, having a test on macOS, i.e., having two tests is better simply because of higher coverage.